src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,111 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double PERCENTAGE_ARRAY_GROWTH = 1.5D;
+    private Object[] elementData;
+    private int size;
+
+    public ArrayList(int initCapacity) {
+        if (initCapacity <= 0) {
+            throw new IllegalArgumentException();
+        }
+        elementData = new Object[initCapacity];
+    }
+
+    public ArrayList() {
+        elementData = new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        resizeIfNeeded();
+        elementData[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        indexExsist(index, size + 1);
+        resizeIfNeeded();
+        System.arraycopy(elementData, index, elementData, index + 1, size - index);
+        elementData[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int index = 0; index < list.size(); index++) {
+            add(list.get(index));
+        }
     }
 
     @Override
+    @SuppressWarnings("unchecked")
     public T get(int index) {
-        return null;
+        indexExsist(index, size);
+        return (T) elementData[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        indexExsist(index, size);
+        elementData[index] = value;
     }
 
     @Override
+    @SuppressWarnings("unchecked")
```

Comment:
Let’s googled about this annotation. Do you really need it?

Reply:
actually, I only need this annotation to eliminate the warnings specified by the compiler. I'll take the annotations, however, does it really need to be taken away?

Reply:
discussed in slack